src/app/shared/request-cache.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@angular/core';
+
+import { HttpRequest, HttpResponse } from '@angular/common/http';
+
+
+export interface RequestCacheEntry {
+  url: string;
+  response: HttpResponse<any>;
+  lastRead: number;
+}
+
+export abstract class RequestCache {
+  abstract get(req: HttpRequest<any>): HttpResponse<any> | undefined;
+  abstract set(req: HttpRequest<any>, response: HttpResponse<any>): void;
+}
+
+const maxAge = 300000; // maximum cache age (ms)
```

Comment:
Why such a large number? This has potential to continue to show stale data. The original cache was put in place only to prevent many duplicate requests going out at the same time.